File: prac/go/var.go
package main

import (
	"errors"
	"fmt"
)

func test() (int, error) {
	return 1, errors.New("bad")
}

func main() {
	var i int
	i = 20
	fmt.Println(i)

	i2 := 2
	fmt.Println(i2)

	a, b, c := 1, 2, 3
	fmt.Println(a, b, c)

	var (
		age  int
		name string
	)
	age = 18
	name = "frnak"
	fmt.Println(age, name)

	_, err := test()
	if err != nil {
		fmt.Println(err)
	}
	const PI = 3.1415926
	r := 2.0
	fmt.Println(2 * PI * r)

	const (
		Unknown = 0
		Female  = 1
		Male    = 2 // 長亮定義不使用不抱錯
	)

	const (
		x int = 16
		y     // 無值同上
		s = "abc"
		z
	)
	fmt.Println("y, z:", y, z)

	// const iota 長亮計數器, iota 表行數
	const (
		a1 = iota
		a2
		a3
		a4
	)
	fmt.Println(a1, a2, a3, a4) // 0 1 2 3

}

File: prac/py/var.py
from collections import namedtuple
from typing import List, Dict


def main():
    print("hello", end="")
    print(" python")

if __name__ == "__main__":
    main()
    my_list = ["frank", "handsome", "575"]
    # 匿名便亮 _
    for _, e in enumerate(my_list):
        print(e)

    # 元祖 不可改
    sex_tuple = ("male", "female")
    # namedtuple 元祖增強

# 3.5 後有類型申明功能，僅提示，不會影響運行
age: int = 12
name: str = "frank"
sex: bool = True
weight: float = 75
x: bytes = b"test"
courses: list =['django', 'scrapy', 'tornado']
courses2: List[str] = ['django', 'scrapy', 'tornado']

user_info:Dict[str, float] = {"frank": 18}